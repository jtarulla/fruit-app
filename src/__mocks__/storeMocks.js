import Vuex from 'vuex';
import { mutations } from '@/store/modules/fruits'

export const state = {
  fruits: [
      {
          isFruit: true,
          name: 'pineapple',
          image: 'https://raw.githubusercontent.com/cycloidio/fictional-eureka/master/assets/pineapple.jpg',
          price: '828.00',
          color: '#1b3148',
          description: 'The pineapple (Ananas comosus) is a tropical plant with an edible fruit and the most economically significant plant in the family Bromeliaceae.',
          taste: 'Incredible',
          expires: '2022-01-25T06:55:57.979Z',
          id: 1
      },
      {
          isFruit: true,
          name: 'orange',
          image: 'https://raw.githubusercontent.com/cycloidio/fictional-eureka/master/assets/orange.jpg',
          price: '911.00',
          color: '#1d7220',
          description: 'The orange is the fruit of various citrus species in the family Rutaceae (see list of plants known as orange); it primarily refers to Citrus × sinensis, which is also called sweet orange, to distinguish it from the related Citrus × aurantium, referred to as bitter orange. The sweet orange reproduces asexually (apomixis through nucellar embryony); varieties of sweet orange arise through mutations.',
          taste: 'Awesome',
          expires: '2021-01-27T02:42:38.684Z',
          id: 2
      }
  ],
};

export const getters = {
  getFruits: jest.fn().mockReturnValue([
      {
        isFruit: true,
        name: 'pineapple',
        image: 'https://raw.githubusercontent.com/cycloidio/fictional-eureka/master/assets/pineapple.jpg',
        price: '828.00',
        color: '#1b3148',
        description: 'The pineapple (Ananas comosus) is a tropical plant with an edible fruit and the most economically significant plant in the family Bromeliaceae.',
        taste: 'Incredible',
        expires: '2022-01-25T06:55:57.979Z',
        id: 1
      },
      {
        isFruit: true,
        name: 'orange',
        image: 'https://raw.githubusercontent.com/cycloidio/fictional-eureka/master/assets/orange.jpg',
        price: '911.00',
        color: '#1d7220',
        description: 'The orange is the fruit of various citrus species in the family Rutaceae (see list of plants known as orange); it primarily refers to Citrus × sinensis, which is also called sweet orange, to distinguish it from the related Citrus × aurantium, referred to as bitter orange. The sweet orange reproduces asexually (apomixis through nucellar embryony); varieties of sweet orange arise through mutations.',
        taste: 'Awesome',
        expires: '2021-01-27T02:42:38.684Z',
        id: 2
      }
  ]),
  getLoadingStatus: jest.fn().mockReturnValue()
};

export const actions = {
    fetchAllFruits: jest.fn(),
    addFruit: jest.fn(),
    deleteFruit: jest.fn(),
};

export function __createMocks(custom = { getters: {}, mutations: {}, actions: {}, state: {} }) {
  const mockGetters = Object.assign({}, getters, custom.getters);
  const mockMutations = Object.assign({}, mutations, custom.mutations);
  const mockActions = Object.assign({}, actions, custom.actions);
  const mockState = Object.assign({}, state, custom.state);

  return {
    getters: mockGetters,
    mutations: mockMutations,
    actions: mockActions,
    state: mockState,
    store: new Vuex.Store({
      getters: mockGetters,
      mutations: mockMutations,
      actions: mockActions,
      state: mockState,
    }),
  };
}

export const store = __createMocks().store;