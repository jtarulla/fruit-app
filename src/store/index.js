import Vue from 'vue';
import Vuex from 'vuex';
import VuexPersistence from 'vuex-persist';
import localForage from 'localforage';
import Fruits from './modules/fruits';

const vuexLocal = new VuexPersistence({
  storage: localForage,
  asyncStorage: true,
});

Vue.use(Vuex)

export default new Vuex.Store({
  strict: true,
  modules: {
    Fruits
  },
  plugins: [vuexLocal.plugin]
})
