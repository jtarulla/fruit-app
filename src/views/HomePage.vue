<template>
  <div class="home">
    <Header />
    <FruitList />
    <FruitForm
      v-if="showModal"
      @close="showModal = false"
    />
    <div class="home--add">
      <img
          class="home--add-btn"
          src='../assets/plus.png'
          alt="add-new"
          @click="showModal = true"
      />
    </div>
  </div>
</template>

<script>
import FruitList from '@/components/FruitList.vue';
import Header from '@/components/Header.vue';
import FruitForm from '@/components/FruitForm.vue';

export default {
  name: 'Home',
  components: {
    FruitList,
    Header,
    FruitForm
  },
  data(){
    return {
      showModal: false
    }
  }
}
</script>
