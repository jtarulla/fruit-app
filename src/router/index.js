import Vue from 'vue';
import VueRouter from 'vue-router';
import store from '@/store';
import Home from '@/views/HomePage.vue';
import Error from '@/views/ErrorPage.vue';

Vue.use(VueRouter)

export const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  {
    path: '/fruit/:id',
    name: 'Fruit',
    component: () => import(/* webpackChunkName: "fruit-detail" */ '../views/FruitPage.vue')
  },
  {
    path: '*',
    name: 'Error',
    component: Error
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes
})

router.beforeEach(async (to, from, next) => {
  await store.restored;
  next();
});

export default router
