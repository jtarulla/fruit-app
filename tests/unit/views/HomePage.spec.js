import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import router from '@/router'
import HomePage from '@/views/HomePage.vue'
import store from '@/store'
import { __createMocks as createStoreMocks } from '@/__mocks__/storeMocks'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('HomePage view', () => {
  let wrapper

  let storeMocks
  
  beforeEach(async () => {
    storeMocks = createStoreMocks();
    
    await Vue.nextTick();

    wrapper = mount(HomePage, {
        localVue,
        router,
        stubs: {
            RouterLink: RouterLinkStub
        },
        store: storeMocks.store
    })
  })

  it('renders the view', () => {
    expect(wrapper.vm.$options.name).toMatch('Home')
    expect(wrapper.html()).toMatchSnapshot();
  })

  it('dispatches "addFruit" correctly', async () => {
    await wrapper.setData({ showModal: true })
  
    await wrapper.find('form').trigger('submit.prevent')

    expect(storeMocks.actions.addFruit).toHaveBeenCalled()
  })

  it('commit "SET_ONE_FRUIT" and changes the state', () => {
    const fruit = {
        isFruit: true,
        name: "banana",
        image: "https://raw.githubusercontent.com/cycloidio/fictional-eureka/master/assets/banana.jpg",
        price: "866.00",
        color: "#797e4a",
        description: "A banana is an elongated, edible fruit – botanically a berry – produced by several kinds of large herbaceous flowering plants in the genus Musa.",
        taste: "Ergonomic",
        expires: "2022-04-19T19:57:31.521Z",
        id: 3
    }

    expect(store.state.Fruits.fruits).toEqual([])

    store.commit('SET_ONE_FRUIT', fruit)

    expect(store.state.Fruits.fruits).toEqual([fruit])
  })
})