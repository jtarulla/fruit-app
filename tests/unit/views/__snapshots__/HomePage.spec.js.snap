// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage view renders the view 1`] = `
<div class="home">
  <header class="header">
    <div class="container"><a><img src="../assets/logo.png" alt="logo" class="header--logo"></a></div>
  </header>
  <div class="fruit-list">
    <div class="container">
      <div class="fruit-list--content">
        <div class="fruit-card">
          <div class="fruit-card--img"><img src="https://raw.githubusercontent.com/cycloidio/fictional-eureka/master/assets/pineapple.jpg" alt="pineapple"></div>
          <div class="fruit-card--title">
            <div class="fruit-card--title-text">
              pineapple
            </div>
            <div class="fruit-card--title-delete"><svg data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 50 50" role="img" aria-labelledby="icon-icon-close-description" class="icon icon-icon-close">
                <title id="icon-icon-close-description">icon close</title>
                <path d="M41.75 5L25 21.75 8.25 5 5 8.25 21.75 25 5 41.75 8.25 45 25 28.25 41.75 45 45 41.75 28.25 25 45 8.25 41.75 5z" data-name="SVG/Close" class="cls-1"></path>
              </svg></div>
          </div> <a>
            <div class="fruit-card--view-more">
              details
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 11.67 19.8">
                <g id="Layer_2" data-name="Layer 2">
                  <g id="Layer_1-2" data-name="Layer 1">
                    <path d="M0,18,1.77,19.8l9.9-9.9L1.77,0,0,1.77,8.13,9.9Z"></path>
                  </g>
                </g>
              </svg>
            </div>
          </a>
        </div>
        <div class="fruit-card">
          <div class="fruit-card--img"><img src="https://raw.githubusercontent.com/cycloidio/fictional-eureka/master/assets/orange.jpg" alt="orange"></div>
          <div class="fruit-card--title">
            <div class="fruit-card--title-text">
              orange
            </div>
            <div class="fruit-card--title-delete"><svg data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 50 50" role="img" aria-labelledby="icon-icon-close-description" class="icon icon-icon-close">
                <title id="icon-icon-close-description">icon close</title>
                <path d="M41.75 5L25 21.75 8.25 5 5 8.25 21.75 25 5 41.75 8.25 45 25 28.25 41.75 45 45 41.75 28.25 25 45 8.25 41.75 5z" data-name="SVG/Close" class="cls-1"></path>
              </svg></div>
          </div> <a>
            <div class="fruit-card--view-more">
              details
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 11.67 19.8">
                <g id="Layer_2" data-name="Layer 2">
                  <g id="Layer_1-2" data-name="Layer 1">
                    <path d="M0,18,1.77,19.8l9.9-9.9L1.77,0,0,1.77,8.13,9.9Z"></path>
                  </g>
                </g>
              </svg>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
  <!---->
  <div class="home--add"><img src="../assets/plus.png" alt="add-new" class="home--add-btn"></div>
</div>
`;
